src/scrolltext.cpp

```diff
@@ -0,0 +1,134 @@
+#include "scrolltext.h"
+
+
+ScrollText::ScrollText(QWidget *parent) :
+    QWidget(parent), scrollPos(0)
+{
+    staticText.setTextFormat(Qt::PlainText);
+
+//    setFixedHeight(fontMetrics().height()*2); //The theme sets this
+    leftMargin = height() / 3;
+
+    setSeparator("   ---   ");
+
+    connect(&timer, SIGNAL(timeout()), this, SLOT(timer_timeout()));
+    timer.setInterval(50);
+}
+
+QString ScrollText::text() const
+{
+    return _text;
+}
+
+void ScrollText::setText(QString text)
+{
+    _text = text;
```

Comment:
One of my biggest pet peeves about putting down text character by character is the line wrapping. When a word is too long, it gets wrapped to the next line while it's being typed out, which is annoying. I wonder if it's possible to predict line wrapping so that a newline can be placed ahead of time.

Reply:
Maybe, this piece of code was simply adapted from older Qt version by me, I didn't delve too deeply into it